### PR TITLE
Change tests to allow different k8s releases

### PR DESCRIPTION
**Pull request by TerryHowe**
Standardized environment variables and parameters. Allowed multiple release support and build releases. Added a preflight check:

```
% ./run_all.sh 
This script will create a cluster, run tests and tear it down
export KOPS_STATE_STORE=s3://kops-state-store-hrxjdwtdkgvxxsuychrauvk
✔ AWS_REGION=us-west-2
✔ AWS CLI authenticated
✔ KOPS_STATE_STORE=s3://kops-state-store-hrxjdwtdkgvxxsuychrauvk
✔ KOPS_CLUSTER_NAME=tlhowe.example.com
Kops executable ./kops-1-18 available...
...
```

### CODE REVIEW

File: development/kops/set_environment.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    export BUCKET_NAME=${KOPS_STATE_STORE#"s3://"}
+
+    if [ -z "$KOPS_CLUSTER_NAME" ]; then
+        COOL=false
+        echo -e "${BAD} KOPS_CLUSTER_NAME must be set and exported"
+    else
+        echo -e "${GOOD} KOPS_CLUSTER_NAME=${KOPS_CLUSTER_NAME}"
+    fi
+fi
+export COOL
```
Comment (jaxesn) [inline development/kops/set_environment.sh:68]: Is there ever a time when calling this script where you dont check for $COOL and exit if not?  Would it be better to just exit here?  nit: is there a better variable name to use if we are using it across scripts like this?

File: development/kops/set_environment.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+VERSION=$(cat ${BASEDIR}/../../projects/kubernetes/kubernetes/${RELEASE_BRANCH}/GIT_TAG)
+export DEFAULT_REPOSITORY_URI=public.ecr.aws/eks-distro
+export REPOSITORY_URI=${REPOSITORY_URI:-${DEFAULT_REPOSITORY_URI}}
+export CNI_VERSION_URL=https://distro.eks.amazonaws.com/kubernetes-${RELEASE_BRANCH}/releases/${RELEASE}/artifacts/plugins/v0.8.7/cni-plugins-linux-amd64-v0.8.7.tar.gz
```
Comment (jaxesn) [inline development/kops/set_environment.sh:73]: We can change this later, but this .8.7 version would change whenever we bump the cni plugins project version.

**Review by jaxesn — CHANGES_REQUESTED** [review]
This looks pretty good, using the set_env script seems to clean some of this up for us!  Im not in love with the `$COOL || exit 1` stuff, but could be convinced otherwise.

File: development/kops/set_environment.sh (hunk 2 above)
Comment (TerryHowe) [reply development/kops/set_environment.sh:73]: Yeh, we should probably make a change someday that pulls this either out of the website we are going to stand up or out of the CRD.

File: development/kops/set_environment.sh (hunk 1 above)
Comment (TerryHowe) [reply development/kops/set_environment.sh]: Done. It is now PREFLIGHT_CHECK_PASSED

File: development/kops/create_values_yaml.sh [hunk 3]
```diff
@@ -16,54 +16,42 @@
...
-                  --repository-name "${REPOSITORY_NAME}" | \
-                  jq -r "${QUERY}")
-    else
-        #
-        # Private repo
-        #
-        QUERY='[.imageDetails[] | select(.imageTags != null)] | sort_by(.imagePushedAt)|reverse|first|.imageTags[0]'
-        TAG=$(aws ecr  describe-images --repository-name "${REPOSITORY_NAME}" | \
-                  jq -r "${QUERY}")
+        if aws --region us-east-1 ecr-public  describe-images --repository-name "${REPOSITORY_NAME}" --image-ids=imageTag=${DEFAULT_TAG}-${RELEASE} 2>/dev/null >/dev/null
```
Comment (abhay-krishna) [inline development/kops/create_values_yaml.sh:42]: nit: extra space
```suggestion
        if aws --region us-east-1 ecr-public describe-images --repository-name "${REPOSITORY_NAME}" --image-ids=imageTag=${DEFAULT_TAG}-${RELEASE} 2>/dev/null >/dev/null
```

**Review by jaxesn — APPROVED** [review]
/lgtm

File: development/kops/create_values_yaml.sh [hunk 4]
```diff
@@ -16,54 +16,42 @@
...
+        then
+            TAG=${DEFAULT_TAG}-${RELEASE}
+        else
+            TAG=${DEFAULT_TAG}-${DEFAULT_RELEASE}
+        fi
+    else # Private
+        if aws ecr  describe-images --repository-name "${REPOSITORY_NAME}" --image-ids=imageTag=${DEFAULT_TAG}-${RELEASE} 2>/dev/null >/dev/null
```
Comment (abhay-krishna) [inline development/kops/create_values_yaml.sh:49]: nit: extra space, same as above

**Pull request merged**